test/integration/component/maint/test_vpc.py

```diff
@@ -0,0 +1,400 @@
+        cls.zone = get_zone(cls.api_client, cls.testClient.getZoneForTests())
+        cls.template = get_template(
+            cls.api_client,
+            cls.zone.id,
+            cls.services["ostype"]
+        )
+        cls.services["virtual_machine"]["zoneid"] = cls.zone.id
+        cls.services["virtual_machine"]["template"] = cls.template.id
+
+        cls.service_offering = ServiceOffering.create(
+            cls.api_client,
+            cls.services["service_offering"]
+        )
+        cls.vpc_off = VpcOffering.create(
+            cls.api_client,
+            cls.services["vpc_offering"]
+        )
+        cls.vpc_off.update(cls.api_client, state='Enabled')
+        cls._cleanup = [
+            cls.service_offering,
+        ]
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            # Cleanup resources used
+            cleanup_resources(cls.api_client, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.account = Account.create(
+            self.apiclient,
+            self.services["account"],
+            admin=True,
+            domainid=self.domain.id
+        )
+        self.cleanup = []
+        self.cleanup.insert(0, self.account)
+        if self.unsupportedHypervisor:
+            self.skipTest("not supported on %s" % self.hypervisor)
+        return
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            self.debug("Warning: Exception during cleanup : %s" % e)
+        return
+
+    def updateConfigurAndRestart(self, name, value):
+        Configurations.update(self.apiclient, name, value)
+        self.RestartServers()
+        time.sleep(self.services["sleep"])
+
+    def RestartServers(self):
+        """ Restart management
+        server and usage server """
+        sshClient = SshClient(
+            self.mgtSvrDetails["mgtSvrIp"],
+            22,
+            self.mgtSvrDetails["user"],
+            self.mgtSvrDetails["passwd"]
+        )
+        command = "service cloudstack-management restart"
+        sshClient.execute(command)
+        return
+
+    @attr(tags=["advanced", "intervlan", "dvs", "test"], required_hardware="true")
+    def test_01_create_tier_Vmxnet3(self):
+        """
+            Test to create vpc tier with nic type as Vmxnet3
+            #1.Set global setting parameter "vmware.systemvm.nic.device.type"
+            to "Vmxnet3"
+            #2.Create VPC
+            #3.Create one tier
+            #4.Deploy one guest vm in the tier created in step3
+        """
+        if self.hypervisor.lower() not in ['vmware']:
+            self.skipTest("This test can only run on vmware setup")
```

Comment:
@DaanHoogland @swill to run this test, you'll need a vmware environment. I read the code, LGTM.

Looking at the test this is testing that a particular nic type works in a VPC, than test VPC itself. Ideally, we would want few tests for VPC that can be run against any of the major 3 hypervisors (xen, vmware, kvm) but it's a good start.